File: app/src/main/java/org/kexie/android/ftper/widget/GenericQuickAdapter.java
package org.kexie.android.ftper.widget;

import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

@SuppressWarnings("WeakerAccess")
public class GenericQuickAdapter<X>
        extends BaseQuickAdapter<X, GenericQuickAdapter.GenericViewHolder> {

    protected final int mName;

    public GenericQuickAdapter(int layoutResId, int name) {
        super(layoutResId);
        mName = name;
        openLoadAnimation();
    }

    @Override
    protected void convert(GenericViewHolder helper, X item) {
        helper.getBinding().setVariable(mName, item);
    }

    public static class GenericViewHolder extends BaseViewHolder {
        private ViewDataBinding mBinding;
        public GenericViewHolder(View view) {
            super(view);
            mBinding = DataBindingUtil.bind(view);
        }

        @SuppressWarnings("unchecked")
        public <T extends ViewDataBinding> T getBinding() {
            return (T) mBinding;
        }
    }
}

File: app/src/main/java/org/kexie/android/ftper/widget/GenericQuickAdapter.kt
package org.kexie.android.ftper.widget

import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder

//ViewModel拥有数据的控制权
class GenericQuickAdapter<X>(layoutId: Int,
                             private val mName: Int)
    : BaseQuickAdapter<X, GenericQuickAdapter.GenericViewHolder>(layoutId) {

    init {
        openLoadAnimation()
    }

    override fun convert(helper: GenericViewHolder, item: X) {
        helper.mBinding.setVariable(mName, item)
    }
    class GenericViewHolder(view: View) : BaseViewHolder(view) {
        val mBinding: ViewDataBinding = DataBindingUtil.bind(view)!!
    }
}